packages/core/src/status.ts

```diff
@@ -38,9 +38,19 @@ export async function status(
   for (const [futureId, deployedContract] of Object.entries(
     deployedContracts
   )) {
-    const artifact = await artifactResolver.loadArtifact(
-      deployedContract.contractName
-    );
+    let artifact: Artifact<Abi>;
+
+    try {
+      artifact = await artifactResolver.loadArtifact(
+        deployedContract.contractName
+      );
+    } catch (e) {
+      if (e instanceof Error && /HH700/g.test(e.message)) {
+        artifact = await deploymentLoader.loadArtifact(deployedContract.id);
```

Comment:
We could instead always call the deploymentLoader as the status is about a deployment, and the artifacts are gonna be stored already.

Reply:
We can change `artifactResolver: Omit<ArtifactResolver, "getBuildInfo">` to `_artifactResolver?: Omit<ArtifactResolver, "getBuildInfo">` so we keep it backward compatible and also indicate that it won't be used.